docs/hlo_dumps.md

```diff
@@ -0,0 +1,219 @@
+
+## Google Colab
+
+### Environment variables
+
+In the first executed cell of your notebook (because environment variables and command-line flags are usually only processed once, e.g., at module-import time or XLA backend initialization time), add the `XLA_FLAGS` detailed above with `os.environ`, for example:
+
+```python
+import os
+os.environ['XLA_FLAGS'] = "--xla_dump_to=CNS_DIRECTORY_PATH"
+```
+
+This will dump the computation to `CNS_DIRECTORY_PATH`, (example path: `/cns/el-d/home/jaxian/xladump.`)
+
+You can use all the flags mentioned in the Local Execution section.
+
+### JAX-specific options
+
+Similar to local execution; for live, interactive introspection you can directly print a computation’s pre-optimized HLO:
+
+```python
+def f(x):
+    return jax.numpy.sin(jax.numpy.cos(x))
+
+c = jax.jit(f).lower(3.).compiler_ir('hlo')
+
+print(c.as_hlo_text())
+```
+
+You can also directly print a computation’s optimized HLO:
+
+```python
+def optimized_HLO(f, *args, platform=None):
+    print(jax.jit(f).lower(*args).compile().as_text())
+
+def f(x):
+    return jax.numpy.sin(jax.numpy.cos(x))
+
+optimized_HLO(f, 1.0)
+```
+
+#### Dumping All/Small Computations
+
+If you want to see everything in a dump including all small compilations, set the JAX environment variable:
+
+```shell
+JAX_COMPILER_DETAILED_LOGGING_MIN_OPS=0
+```
+
+#### Mosaic
+
+Mosaic is a compiler for the Pallas TPU backend, and the experimental Pallas GPU backend. To dump mosaic computation, set the following flag:
+
+```shell
+--xla_mosaic_dump_to=/tmp/mosaic_dumps
+```
+
+Or, set TPU init arguments as an environment variable:
+
+```shell
+export LIBTPU_INIT_ARGS="--xla_mosaic_dump_to=/tmp/mosaic_dumps"
+```
+
+Check out the [JAX documentation on Pallas and Mosaic](https://docs.jax.dev/en/latest/pallas/index.html) to learn more.
+
+## More with HLO Dumps
+
+### Find the right computation
+
+Usually, many computations get dumped. The dumped files are explicitly named with the JAX, Tensorflow, or PyTorch/XLA "computation name” that are called out in the logs, making it easy to identify the relevant HLO files. For example:
+
+```
+1624325116260738.module_0065.pmap__unnamed_wrapped_function_.186875.before_optimizations.txt
+```
+
+Otherwise, you can use `ripgrep` to quickly identify which module holds particular symbols or computations.
+
+**Tip:** Include the 3 dumped before/after/buffer-assignment files of interest to your bug reports.
+
+### HLO Conversion
+
+Sometimes you may have HLOProto buffer dumps and want readable HLO text, or vice-versa.
+
+A tool called `hlo-opt` that can translate between HLO proto and text formats. Learn to use it: [XLA Tooling documentation: hlo-opt](tools.md#hlo-opt-convert-hlo-module-formats).
+
+### Replay
+
+The dumped computations can be run it on fake data or input snapshots, which is useful as a simple reproducible example for XLA developers.
```

Comment:
This sentence could be phrased better.

The following commands allow you to run the dumped computations on a specified XLA backend. This is a convenient way to iterate and debug issues in XLA.